### DISCUSSION TITLE
Channel naming convention

### DISCUSSION THREAD

**Discussion by del42** [chorus-ai/chorus_waveform#3] [General]
Some of the standard channels require a common naming convention, as some devices use slightly varied names for them.

**Limb Leads**
- **I**: Records the electrical potential difference between the right and left arms.
- **II**: Captures the potential difference between the right arm and left leg.
- **III**: Measures the potential difference between the left arm and left leg.

**Augmented Limb Leads (aVR, aVL, and aVF)**
- **aVR**: (augmented Voltage Right) Concentrates on the right arm.
- **aVL**: (augmented Voltage Left) Focuses on the left arm.
- **aVF**: (augmented Voltage Foot) Targets the left leg.

**Chest (Precordial) Leads**
- **V1**: Placed at the fourth intercostal space on the right sternal border.
- **V2**: Located at the fourth intercostal space on the left sternal border.
- **V3**: Positioned midway between V2 and V4.
- **V4**: Situated on the fifth intercostal space along the midclavicular line.
- **V5**: Aligned horizontally with V4 at the left anterior axillary line.
- **V6**: Set horizontally in line with V4 at the left midaxillary line.

**Other Leads**
- **SpO2**: Measured using a pulse oximeter, a non-invasive device that clips onto a patient's finger, toe, or earlobe.
- **RR**: Typically refers to "Respiratory Rate," which is the number of breaths a person takes per minute.
-  **PA**: Often refers to the Pulmonary Artery, the blood vessel that carries deoxygenated blood from the right ventricle of the heart to the lungs for oxygenation.
- **CVP**: Central Venous Pressure is the pressure in the thoracic vena cava near the right atrium of the heart. It reflects the amount of blood returning to the heart and the ability of the heart to pump the blood into the arterial system.

ℹ️ NOTE: This is not yet final standard channel names.

**Comment by clermontg** [comment]
Other frequent channels include ABP (invasive blood pressure), ART (another way to call ABP), and some folks have both simultaneously. ICP (intracranial pressure).  Here is a list of the most common channels we have in our waveform data

label|COUNT(*)
II|8281
Resp|8211
V|8168
aVR|6760
Pleth|6703
III|5375
I|3713
ART|2440
MCL|1951
PlethT|854
ABP|762
nolabel|630
aVF|320
aVL|302
CVP|290
V1|187
AS|183
ES|183
AI|183
ICP|176
UAP|128
CO2|98
PAP|94
PLETHl|76

**Reply by clermontg** [reply]
the second number is the number of encounters where the signal is present, out of 8281 encounters

**Comment by birm** [comment]
We've been working on some analysis of the wfdb files we have at Emory. Some of which have odd invalid fields (such as "Deg C") due to whitespace in the channel name and were excluded from our analysis, but here are some of the other fields we found, including calculated/derived fields:

```
ABP
ABP-D
ABP-M
ABP-S
AI
AMP
Ao
AO-D
AO-M
AO-S
APNEA
APRV-HI-T
APRV-LO-T
AR1
AR1-D
AR1-M
AR1-R
AR1-S
AR2
AR2-D
AR2-M
AR2-R
AR2-S
AR3
AR3-D
AR3-M
AR3-R
AR3-S
AR4
AR4-D
AR4-M
AR4-R
AR4-S
AR5
AR5-D
AR5-M
AR5-R
AR5-S
AR6
AR6-D
AR6-M
AR6-R
AR6-S
AR7
AR7-D
AR7-M
AR7-R
AR7-S
AR8-R
ART
ART-D
ART-M
ART-S
AS
AVF
aVF
aVL
AVL
AVR
aVR
AWRR
BIS
Bpm
BrMin
BT
CI
CO
CO₂
CO2
CO2-EX
CO2-IN
CO2-RR
CPP
CPP1
CPP2
CPP3
CPP4
CPP5
CPP6
CPP7
CPP8
CUFF
CVP
CVP-M
CVP1
CVP2
CVP3
CVP4
CVP5
CVP6
CVP7
CVP8
d
dB
DELTA-TMP
dQTc
ECT
EMG
ES
EtCO2
FAP
FE1-D
FE1-M
FE1-R
FE1-S
FE2-D
FE2-M
FE2-R
FE2-S
FE3-D
FE3-M
FE3-R
FE3-S
FE4-D
FE4-M
FE4-R
FE4-S
FE5-D
FE5-M
FE5-R
FE5-S
FE6-D
FE6-M
FE6-R
FE6-S
FiO2
FLW-R
HR
Hz
I
I:E
IC1
IC2
ICG
ICP
ICP-M
ICP1
ICP2
ICP3
ICP4
ICP5
ICP6
ICP7
ICP8
II
III
IMCO2
IN_HLD
INSP-TM
IT
LA1
LA2
LA3
LA4
LA6
LAP
LAP-M
MAWP
MCL
MEAS-PEEP
mm
mmHg
MV
NBP-D
NBP-M
NBP-S
O2-EXP
O2-INSP
P
PA1
PA1-D
PA1-M
PA1-S
PA2
PA2-D
PA2-M
PA2-S
PA3
PA3-D
PA3-M
PA3-S
PA4
PA4-D
PA4-M
PA4-S
PA5-D
PA5-M
PA5-S
PA6
PA6-D
PA6-M
PA6-S
PAP
PAP-D
PAP-M
PAP-S
Parris
PAW
PAWP
PEEP
PIP
Pleth
PLETHl
PlethT
PLTHpo
PLTHpr
PT-RR
PULSE
PVC
QTc
RA1
RA2
RA3
RA4
RAP
RAP-M
Rate
RESIS
RESP
Resp
RHY
RM_FLOW
RM_VOL
RR
SEF
SP1
SP2
SP3
SP4
SP5
SP6
SP8
SPO2
SpO2_l
SPO2-%
SPO2-R
SPO2T
SQI
SR
ST-AVF
ST-AVL
ST-AVR
ST-I
ST-II
ST-III
ST-MCL
ST-V
ST-V1
ST-V2
ST-V3
ST-V4
ST-V5
ST-V6
STindx
SvO2
T-ART
T-BLOOD
T-CORE
T-ESOPH
T-RECT
T-SKIN
TEMP
TMP-1
TMP-2
TV
U-TEMP
U-VOL
UAC1
UAC1-D
UAC1-M
UAC1-R
UAC1-S
UAC2
UAC2-D
UAC2-M
UAC2-R
UAC2-S
UAC3
UAC3-D
UAC3-M
UAC3-R
UAC3-S
UAC4
UAC4-D
UAC4-M
UAC4-R
UAC4-S
UAP
UAP-D
UAP-M
UAP-S
UVC1
UVC2
UVC3
UVC4
UVP
UVP-M
V
V1
V2
V3
V4
V5
V6
Vent_Rate
Vent-Rate
VNT_FLOW
VNT_PRES
```

**Reply by del42** [reply]
**Waveform**: 
UVC3, SP1, RESP, RM, AVR, AR1, UAC2, AR6, UAC3, ICP8, RAP, PA2, V5, AR7, UAC1, ICP, PA4, CVP3, UVC4, V3, P, ICP6, AR3, SP8, ICP1, UVC1, Ao, RA2, VNT, V, aVR, AS, FAP, AI, LA4, V2, ABP, SPO2, ICP3, Resp, CVP2, Pleth, SP3, LA6, IC2, LA2, CO₂, AR4, CVP8, ART, III, ES, ICP5, CVP6, UVP, RM, RA3, RA4, PLTHpr, MCL, II, V4, CVP4, aVL, SP2, CO2, UAP, LAP, RR, AVL, AR5, SP6, ICP4, AR2, CVP5, I, PlethT, PA3, UVC2, ICP2, LA1, SP5, PLTHpo, PLETHl, PA1, PAP, PA6, VNT, SvO2, LA3, AVF, V1, UAC4, ICG, CVP1, CVP, SP4, RA1, aVF, ICP7, V6, IC1

**Numerics**: 
TEMP, CUFF, FE1-R, UVC3, ST-AVF, SP1, UAP-D, UAC4-S, APNEA, RESP, AWRR, FE2-D, FE4-D, ABP-M, T-CORE, FE3-S, ICP-M, UAC2-M, AO-S, ICP8, FE6-S, APRV-LO-T, PA2-S, UAC1-D, CPP6, UAC3-S, PA6-M, %, FE6-M, ?, Parris, CVP3, SPO2T, mmHg, PA4-S, UVC4, ST-V2, INSP-TM, ICP6, RHY, TMP-2, ST-V5, ST-AVR, SP8, AR7-M, ICP1, NBP-S, ST-V, UVC1, CO2-RR, FE1-D, IT, AR8-R, Vent, RA2, CVP-M, CVP7, ST-II, UVP-M, UAC1-R, PIP, DELTA-TMP, Bpm, AR2-R, PA6-S, STindx, AR5-S, AR4-S, PA4-M, BIS, AR6-R, dQTc, SPO2-%, AR5-D, ART-D, CPP4, FE6-R, UAC4-M, U-VOL, FE5-D, CPP5, TMP-1, PULSE, AR7-R, AO-D, PT-RR, SQI, UAC3-D, CPP3, AR6-M, PA2-M, AR4-D, AMP, LA4, UAC2-S, FE2-R, AR7-D, SPO2, PAW, ICP3, AR5-M, BrMin, MV, CVP2, LAP-M, ABP-S, CPP2, T-ART, SP3, LA6, T-ESOPH, IC2, LA2, AR4-R, PEEP, FE5-S, FE2-M, SEF, FE1-S, CVP8, T-RECT, AR3-M, PVC, NBP-M, NBP-D, IMCO2, AR5-R, IN, HR, AR2-M, O2-INSP, AR2-D, FE6-D, ICP5, CVP6, UAC2-R, PA3-D, CPP, PA3-M, AR6-D, UAC1-S, PA2-D, RA3, RA4, PAWP, d, UAC4-D, CPP1, EtCO2, FE4-S, PAP-M, T-SKIN, UAC4-R, CVP4, FE2-S, FE5-M, UAC2-D, SpO2, PA1-M, Vent-Rate, AR7-S, SP2, U-TEMP, SPO2-R, I:E, ST-III, PA5-D, T-BLOOD, FE5-R, RR, ST-V6, ST-AVL, FE4-M, AO-M, AR3-D, PA1-S, ST-V3, UAC1-M, AR1-S, SP6, ST-I, PA6-D, FE3-R, CPP7, ST-V1, ICP4, RESIS, O2-EXP, FE1-M, Rate, RAP-M, UAC3-R, CVP5, ART-M, CI, FE4-R, PAP-D, FE3-D, FiO2, PA4-D, UVC2, PA5-S, ICP2, LA1, AR4-M, SP5, mm, CPP8, ART-S, CO2-IN, MAWP, AR1-D, ECT, UAP-M, APRV-HI-T, SR, ABP-D, UAC3-M, AR6-S, AR3-R, ST-MCL, CO2-EX, AR2-S, PA3-S, Hz, BT, AR3-S, LA3, PA1-D, PA5-M, AR1-M, UAP-S, EMG, FLW-R, ST-V4, CVP1, PAP-S, dB, FE3-M, SP4, QTc, RA1, CO, ICP7, MEAS-PEEP, AR1-R, IC1, TV

**Either**: 
LA6, RA2, SP3, SP1, LA2, RA4, ICP7, SP4, CVP1, RA1, LA4, ICP2, ICP8, IC2, CVP4, CVP2, IC1, CVP8, UVC1, CVP6, ICP5, CVP3, ICP1, ICP6, SPO2, SP5, RESP, UVC3, SP8, ICP3, UVC2, LA3, CVP5, SP6, UVC4, SP2, LA1, RR, ICP4, RA3

**Comment by bemoody** [comment]
Waveform and numerics channels in MIMIC-IV:

https://raw.githubusercontent.com/MIT-LCP/mimic-code/main/mimic-iv/concepts/concept_map/waveforms-summary.csv

https://raw.githubusercontent.com/MIT-LCP/mimic-code/main/mimic-iv/concepts/concept_map/numerics-summary.csv

**Comment by wa6gz** [comment]
Waveforms (name, sample rate, file count)
```
AVL	240.0	3992
AVR	240.0	3764
AVF	240.0	3715
II	240.0	3680
I	240.0	2594
III	240.0	2192
V	240.0	1330
V2	240.0	9
V3	240.0	4
V4	240.0	1
V5	240.0	1
ART1	120.0	634
UAC1	120.0	280
CVP2	120.0	185
ART2	120.0	144
PA4	120.0	86
FEM3	120.0	66
ICP2	120.0	29
ART3	120.0	20
CVP1	120.0	16
SP4	120.0	15
SP1	120.0	12
UAC2	120.0	12
FEM1	120.0	9
SP2	120.0	8
ICP1	120.0	8
CVP4	120.0	7
FEM2	120.0	7
ART4	120.0	6
PA2	120.0	5
CVP3	120.0	5
SP3	120.0	5
PA1	120.0	3
PA3	120.0	2
FEM4	120.0	2
UVC2	120.0	2
SP7	120.0	1
LA2	120.0	1
SPO2	60.0	3819
```

and 0.5 Hz Numerics (name, file count):

```
HR        	3814
SPO2_pct  	3793
SPO2_R    	3790
RESP      	3767
NBP_D     	3706
NBP_M     	3706
NBP_S     	3706
PVC       	3669
ST_II     	3640
CUFF      	3563
ST_I      	2471
ST_III    	2076
ST_AVF    	1296
ST_AVL    	1296
ST_AVR    	1296
ST_V1     	1295
O2_INSP   	790
CO2_EX    	672
ART1_D    	634
ART1_M    	634
ART1_S    	634
ART1_R    	631
CO2_IN    	408
O2_EXP    	408
CO2_RR    	391
TMP_1     	364
UAC1_D    	278
UAC1_M    	278
UAC1_R    	278
UAC1_S    	278
rMV       	188
SPO2M_R   	183
SPO2M_pct 	183
rMVs      	179
rPIP      	150
CVP2      	149
rCDYN     	121
ART2_D    	111
ART2_M    	111
ART2_S    	111
TMP_2     	105
ART2_R    	104
PA4_D     	87
PA4_M     	87
PA4_S     	87
rI:E      	83
FEM3_D    	66
FEM3_R    	66
FEM3_M    	65
FEM3_S    	65
BT        	56
IT        	56
CO        	54
CPP2      	29
ICP2      	29
SvO2      	27
CBT       	26
BIS       	26
EMG       	26
SQI       	26
SR        	26
CCI       	24
CCO       	24
CSVR      	24
CSVRI     	24
ST_V2     	20
ST_V3     	17
SP1       	16
ST_V6     	15
CVP1      	15
ART3_D    	15
ART3_M    	15
ART3_R    	15
ART3_S    	15
ST_V5     	15
UAC2_D    	13
UAC2_M    	13
UAC2_S    	13
ST_V4     	13
rMAWP     	13
UAC2_R    	12
FEM1_D    	9
FEM1_M    	9
FEM1_R    	9
FEM1_S    	9
SP2       	8
CPP1      	8
ICP1      	8
CVP4      	7
ART4_D    	7
ART4_M    	7
ART4_S    	7
PA2_D     	6
PA2_M     	6
PA2_S     	6
FEM2_D    	6
FEM2_M    	6
FEM2_S    	6
ART4_R    	6
PAW       	5
SP4       	5
FEM2_R    	5
CVP3      	5
SP3       	4
rRR       	3
FEM4_D    	3
FEM4_M    	3
FEM4_R    	3
FEM4_S    	3
PA1_D     	3
PA1_M     	3
PA1_S     	3
PA3_D     	2
PA3_M     	2
PA3_S     	2
UVC2      	2
RA1       	1
UVC1      	1
SP7       	1
LA2       	1
```

**Comment by edilbertoamorim** [comment]
For UCSF on a sample of 68 cases, this is what we got: 

Total Signals Count: 

{'II': 68, 'Pleth': 68, 'Resp': 68, 'V': 68, 'aVR': 68, 'ART': 58, 'I': 55, 
'III': 55, 'CVP': 30, 'CO2': 17, 'ICP': 12, 'PAP': 6, 'ABP': 6, 'LAP': 4, 'AWP': 4, 'AWF': 4, 'IC1': 3, 
'RAP': 3, 'BAP': 2, 'Ao': 2, 'P1': 2, 'IC2': 1, 'ES': 1, 'AI': 1, 'AS': 1, 'V1': 1, 'PLETHl': 1, 'P': 1, 'aVL': 1, 'UAP': 1})
